### PR TITLE
Use domain name as organization name

### PR DESCRIPTION
**Pull request by alexopenline**
## Proposed changes
Use domain name as organisation name

## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Summary by CodeRabbit

- **New Features**
  - Introduced a new utility to extract the first part of a string before a given delimiter.

- **Refactor**
  - Updated event handling methods to use a new function for setting span tags and log fields.
  - Enhanced logging functionality to use JSON formatting for events.

- **Bug Fixes**
  - Improved the `webScrapeOrganization` function to handle cases with empty organization names by utilizing the domain name from the URL.

- **Tests**
  - Expanded test cases for the new string utility function to ensure robustness, including edge cases.

### CODE REVIEW

**Review by grokkos — DISMISSED** [review]

File: packages/server/customer-os-common-module/utils/string_utils_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	// Edge case: empty string
+	str = ""
+	delimiter = ","
+	expected = ""
+	result = ExtractFirstPart(str, delimiter)
+	if result != expected {
+		t.Errorf("Expected %s, but got %s", expected, result)
+	}
+
+	// Edge case: no delimiter found
```
Comment (max-openline) [inline packages/server/customer-os-common-module/utils/string_utils_test.go]: isn't this the same as the second test case?

File: packages/server/events-processing-platform/subscriptions/graph/location_event_handlers.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ func (h *GraphLocationEventHandler) OnLocationCreate(ctx context.Context, evt ev
 func (h *GraphLocationEventHandler) OnLocationUpdate(ctx context.Context, evt eventstore.Event) error {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "GraphLocationEventHandler.OnLocationUpdate")
 	defer span.Finish()
-	setCommonSpanTagsAndLogFields(span, evt)
+	setEventSpanTagsAndLogFields(span, evt)
```
Comment (max-openline) [inline packages/server/events-processing-platform/subscriptions/graph/location_event_handlers.go:38]: are these changes related to the fallback from org name to domain name?

File: packages/server/events-processing-platform/subscriptions/organization/organization_event_handler.go [hunk 3]
```diff
@@ -195,6 +198,21 @@ func (h *organizationEventHandler) webScrapeOrganization(ctx context.Context, te
 	return nil
 }
 
+func (h *organizationEventHandler) updateOrganizationNameIfEmpty(ctx context.Context, tenant, url string, organization *entity.OrganizationEntity, span opentracing.Span) {
+	if organization.Name == "" && strings.Contains(url, ".") {
```
Comment (max-openline) [inline packages/server/events-processing-platform/subscriptions/organization/organization_event_handler.go:202]: although I think the code will work as intended, this part would fail if for any reason we had a bug or failure and got `www.subdomain.domain.com` or `www.domain` for a value rather than `domain.com`, wouldn't perhaps a more thorough check help avoid this? (e.g.: checking it has two parts only, or `.com` at the end, or that there are no subdomain or `www` ?)

**Review by max-openline — APPROVED** [review]
